Apps/NotSensorLogger2/app/src/main/java/com/example/notsensorlogger2/gpsfunctionality/MainActivity2.java

```diff
@@ -0,0 +1,180 @@
+        }
+    };
+
+    /**
+     * Listener for changes in the position of the location indicator.
+     * Updates the camera position of the map view to center on the new location and adjusts the focal point of gesture interactions.
+     */
+    private final OnIndicatorPositionChangedListener onIndicatorPositionChangedListener = new OnIndicatorPositionChangedListener() {
+        @Override
+        public void onIndicatorPositionChanged(@NonNull Point point) {
+            mapView.getMapboxMap().setCamera(new CameraOptions.Builder().center(point).zoom(20.0).build());
+            getGestures(mapView).setFocalPoint(mapView.getMapboxMap().pixelForCoordinate(point));
+        }
+    };
+
+    /**
+     * Listener for map movement gestures.
+     * Removes location tracking listeners and shows the floating action button when map movement begins.
+     */
+    private final OnMoveListener onMoveListener = new OnMoveListener() {
+        @Override
+        public void onMoveBegin(@NonNull MoveGestureDetector moveGestureDetector) {
+            getLocationComponent(mapView).removeOnIndicatorBearingChangedListener(onIndicatorBearingChangedListener);
+            getLocationComponent(mapView).removeOnIndicatorPositionChangedListener(onIndicatorPositionChangedListener);
+            getGestures(mapView).removeOnMoveListener(onMoveListener);
+            floatingActionButton.show();
+        }
+
+        @Override
+        public boolean onMove(@NonNull MoveGestureDetector moveGestureDetector) {
+            return false;
+        }
+
+        @Override
+        public void onMoveEnd(@NonNull MoveGestureDetector moveGestureDetector) {
+
+        }
+    };
+
+    /**
+     * Initializes the activity, inflates the layout, and sets up the map view and its components.
+     * This method requests location permissions if not already granted and sets up the map components
+     * once the style is loaded. It also configures a back button to return to the previous screen.
+     *
+     * @param savedInstanceState If the activity is being re-initialized after previously being shut down,
+     *                           this Bundle contains the data it most recently supplied. Otherwise, it is null.
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main2);
+
+        mapView = findViewById(R.id.mapView);
+        floatingActionButton = findViewById(R.id.focusLocation);
+        Button backButton = findViewById(R.id.backButton);
+
+
+        if (ActivityCompat.checkSelfPermission(MainActivity2.this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            activityResultLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION);
+        }
+
+        floatingActionButton.hide();
+        mapView.getMapboxMap().loadStyleUri("mapbox://styles/yashnalluri/clt6uqpyk00gd01o80tmo62ms", new Style.OnStyleLoaded() {
```

Comment:
personal style uri? If nothing special added to the customized uri, can use the [classic style uri](https://docs.mapbox.com/android/maps/guides/styles/set-a-style/#classic-style-templates-and-custom-styles) from the apk.

Reply:
Okay, change to a classic style